src/warehouses/components/WarehouseDetailsPage/WarehouseDetailsPage.tsx

```diff
@@ -8,32 +8,37 @@ import PageHeader from "@saleor/components/PageHeader";
 import Savebar from "@saleor/components/Savebar";
 import { ShopInfo_shop_countries } from "@saleor/components/Shop/types/ShopInfo";
 import { AddressTypeInput } from "@saleor/customers/types";
+import { WarehouseDetailsFragment } from "@saleor/fragments/types/WarehouseDetailsFragment";
 import { WarehouseErrorFragment } from "@saleor/fragments/types/WarehouseErrorFragment";
 import useAddressValidation from "@saleor/hooks/useAddressValidation";
 import { SubmitPromise } from "@saleor/hooks/useForm";
 import useStateFromProps from "@saleor/hooks/useStateFromProps";
 import { sectionNames } from "@saleor/intl";
 import { Backlink } from "@saleor/macaw-ui";
 import { findValueInEnum, maybe } from "@saleor/misc";
-import { CountryCode } from "@saleor/types/globalTypes";
+import {
+  CountryCode,
+  WarehouseClickAndCollectOptionEnum
+} from "@saleor/types/globalTypes";
 import createSingleAutocompleteSelectHandler from "@saleor/utils/handlers/singleAutocompleteSelectChangeHandler";
 import { mapCountriesToChoices, mapEdgesToItems } from "@saleor/utils/maps";
 import React from "react";
 import { FormattedMessage, useIntl } from "react-intl";
 
-import { WarehouseDetails_warehouse } from "../../types/WarehouseDetails";
 import WarehouseInfo from "../WarehouseInfo";
 import WarehouseZones from "../WarehouseZones";
 
 export interface WarehouseDetailsPageFormData extends AddressTypeInput {
   name: string;
+  isPrivate: boolean;
```

Comment:
It would seem that it's `string` type here; more specifically `"true" | "false"`

Reply:
Done, I have used `string` due to the narrowing problems few lines further with `prototype.boolean.toString()`